bluemira/equilibria/analysis.py

```diff
@@ -0,0 +1,1423 @@
+
+        self.from_cocos = from_cocos
+        self.to_cocos = to_cocos
+        self.qpsi_positive = qpsi_positive
+        self.control = control_coils
+        self.equilibria, self.profiles = self.get_eqs_and_profiles()
+        self.plotting_profiles = MultiEqProfiles()
+        self.fill_plotting_profiles(self.n_points)
+
+    def get_eqs_and_profiles(self):
+        """
+        Use file name, cocos information etc. to get a list of equilibrium
+        and profile objects.
+
+        Returns
+        -------
+        equilibria:
+            List of equlibrium objects
+        profiles:
+            List of profile objects
+        """
+        equilibria = []
+        profiles = []
+        for file, eq_type, dummy, fc, tc, qs in zip(
+            self.equilibrium_paths,
+            self.fixed_or_free,
+            self.dummy_coils,
+            self.from_cocos,
+            self.to_cocos,
+            self.qpsi_positive,
+            strict=False,
+        ):
+            eq = select_eq(
+                file,
+                fixed_or_free=eq_type,
+                dummy_coils=dummy,
+                from_cocos=fc,
+                to_cocos=tc,
+                qpsi_positive=qs,
+            )
+            if isinstance(self.control, Iterable):
+                eq.coilset.control = self.control
+            if isinstance(self.control, CoilType):
+                eq.coilset.control = eq.coilset.get_coiltype(self.control).name
+            equilibria.append(eq)
+            profiles.append(eq.profiles)
+        return equilibria, profiles
+
+    def make_eq_dataclass_list(self, method, *args):
+        """
+        Create a list of dataclasses from a function with
+        equilbrium as the first argument.
+
+        Returns
+        -------
+        results:
+            list of dataclass objects
+        """
+        results = []
+        for eq in self.equilibria:
+            results.append(method(eq, *args))  # noqa: PERF401
+        return results
+
+    def fill_plotting_profiles(self, n_points):
+        """
+        Calculate profile values for plotting. This is done when a MultiEqAnalysis
+        class initiated but can be reset with different n_points using this function.
+
+        Parameters
+        ----------
+        n_points:
+            number of normalised psi points
+
+        """
+        for profile in self.profiles:
+            self.plotting_profiles.add_profile(profile, n_points)
+
+    def physics_info_table(self):
+        """
+        Create a table with the the physics information
+        from all listed Equilbria.
+        Not for use with FixedPlasmaEquilibrium.
+
+        Returns
+        -------
+        table:
+            Table with equilibria physics information.
+
+        """
+        eq_summaries = self.make_eq_dataclass_list(Equilibrium.analyse_plasma)
+        table = make_table(eq_summaries, self.equilibrium_names)
+        print(table)  # noqa: T201
+        return table
```

Comment:
This will be a string which is not hugely accessible for the user, possibly returning eq_summaries is better?